src/languages/snowflake/snowflake.formatter.ts

```diff
@@ -0,0 +1,352 @@
+  'DROP MANAGED ACCOUNT',
+  'DROP MASKING POLICY',
+  'DROP MATERIALIZED VIEW',
+  'DROP NETWORK POLICY',
+  'DROP PIPE',
+  'DROP PROCEDURE',
+  'DROP REPLICATION GROUP',
+  'DROP RESOURCE MONITOR',
+  'DROP ROLE',
+  'DROP ROW ACCESS POLICY',
+  'DROP SCHEMA',
+  'DROP SEQUENCE',
+  'DROP SESSION POLICY',
+  'DROP SHARE',
+  'DROP STAGE',
+  'DROP STREAM',
+  'DROP TAG',
+  'DROP TASK',
+  'DROP USER',
+  'DROP VIEW',
+  'DROP WAREHOUSE',
+  'EXECUTE IMMEDIATE',
+  'EXECUTE TASK',
+  'EXPLAIN',
+  'GET',
+  'GRANT OWNERSHIP',
+  'GRANT ROLE',
+  'INSERT',
+  'LIST',
+  'MERGE',
+  'PUT',
+  'REMOVE',
+  'REVOKE ROLE',
+  'ROLLBACK',
+  'SHOW COLUMNS',
+  'SHOW CONNECTIONS',
+  'SHOW DATABASES',
+  'SHOW DATABASES IN FAILOVER GROUP',
+  'SHOW DATABASES IN REPLICATION GROUP',
+  'SHOW DELEGATED AUTHORIZATIONS',
+  'SHOW EXTERNAL FUNCTIONS',
+  'SHOW EXTERNAL TABLES',
+  'SHOW FAILOVER GROUPS',
+  'SHOW FILE FORMATS',
+  'SHOW FUNCTIONS',
+  'SHOW GLOBAL ACCOUNTS',
+  'SHOW GRANTS',
+  'SHOW INTEGRATIONS',
+  'SHOW LOCKS',
+  'SHOW MANAGED ACCOUNTS',
+  'SHOW MASKING POLICIES',
+  'SHOW MATERIALIZED VIEWS',
+  'SHOW NETWORK POLICIES',
+  'SHOW OBJECTS',
+  'SHOW ORGANIZATION ACCOUNTS',
+  'SHOW PARAMETERS',
+  'SHOW PIPES',
+  'SHOW PRIMARY KEYS',
+  'SHOW PROCEDURES',
+  'SHOW REGIONS',
+  'SHOW REPLICATION ACCOUNTS',
+  'SHOW REPLICATION DATABASES',
+  'SHOW REPLICATION GROUPS',
+  'SHOW RESOURCE MONITORS',
+  'SHOW ROLES',
+  'SHOW ROW ACCESS POLICIES',
+  'SHOW SCHEMAS',
+  'SHOW SEQUENCES',
+  'SHOW SESSION POLICIES',
+  'SHOW SHARES',
+  'SHOW SHARES IN FAILOVER GROUP',
+  'SHOW SHARES IN REPLICATION GROUP',
+  'SHOW STAGES',
+  'SHOW STREAMS',
+  'SHOW TABLES',
+  'SHOW TAGS',
+  'SHOW TASKS',
+  'SHOW TRANSACTIONS',
+  'SHOW USER FUNCTIONS',
+  'SHOW USERS',
+  'SHOW VARIABLES',
+  'SHOW VIEWS',
+  'SHOW WAREHOUSES',
+  'TRUNCATE MATERIALIZED VIEW',
+  'UNDROP DATABASE',
+  'UNDROP SCHEMA',
+  'UNDROP TABLE',
+  'UNDROP TAG',
+  'UNSET',
+  'USE DATABASE',
+  'USE ROLE',
+  'USE SCHEMA',
+  'USE SECONDARY ROLES',
+  'USE WAREHOUSE',
+]);
+
+const reservedSetOperations = expandPhrases(['UNION [ALL]', 'MINUS', 'EXCEPT', 'INTERSECT']);
+
+const reservedJoins = expandPhrases([
+  '[INNER] JOIN',
+  '[NATURAL] {LEFT | RIGHT | FULL} [OUTER] JOIN',
+  '{CROSS | NATURAL} JOIN',
+]);
+
+const reservedPhrases = expandPhrases(['{ROWS | RANGE} BETWEEN']);
+
+export default class SnowflakeFormatter extends Formatter {
+  tokenizer() {
+    return new Tokenizer({
+      reservedCommands,
+      reservedSelect,
+      reservedSetOperations,
+      reservedJoins,
+      reservedPhrases,
+      reservedKeywords: keywords,
+      reservedFunctionNames: functions,
+      stringTypes: ['$$', `''-qq-bs`],
+      identTypes: ['""-qq'],
+      variableTypes: [
+        // for accessing columns at certain positons in the table
+        { regex: '[$][1-9]\\d*' },
+        // identifier style syntax
+        { regex: '[$][_a-zA-Z][_a-zA-Z0-9$]*' },
+      ],
```

Comment:
Are these correctly written?

1. Should be anything like `$number`, where `number > 0`
2. Shoud be anything like `$identifier`, where identifier starts with char `a-Z`, can contain any combination of multiple `$`, `a-Z` and `0-9`

Reply:
Looks fine to me. I'd suggest simplifying the first one to just:

```
[$][0-9]+
```

It's OK if we accept SQL that's not fully correct - the goal is formatting, not validation. It would only pose a problem with `$0` would be some sort of special syntax that we'd like to handle elsewhere.

Reply:
I see your point, but why not just leave it? It's not that complicated, I can rewrite it to 
`[$][1-9][0-9]*`
if that's better readable?

Reply:
Might as well just leave it as is.